Promise/Promise+Extras.swift

```diff
@@ -150,8 +150,17 @@ extension Promise {
     }
 
     public func recover(_ recovery: @escaping (Error) throws -> Promise<Value>) -> Promise<Value> {
+        return recover(type: Error.self, recovery)
```

Comment:
The two functions could be joined into one by setting a default error type in the method signature. Let me know if you'd prefer it like that:

```swift
public func recover<E: Error>(type errorType: E.Type = Error.self, _ recovery: @escaping (E) throws -> Promise<Value>) -> Promise<Value> {
                                                    ^^^^^^^^^^^^^
```

Reply:
Nevermind, the compiler doesn't allow this.

Reply:
This was going to be my first question 😊